src/server/frontend_xwayland/xcb_connection.cpp

```diff
@@ -119,11 +119,10 @@ mf::XCBConnection::Atom::operator xcb_atom_t() const
         // The initial atomic check is faster, but we need to check again once we've got the lock
         if (atom == XCB_ATOM_NONE)
         {
-            auto const reply = xcb_intern_atom_reply(*connection, cookie, nullptr);
+            std::unique_ptr<xcb_intern_atom_reply_t> const reply{xcb_intern_atom_reply(*connection, cookie, nullptr)};
```

Comment:
Is this actually going to do the right thing? This is going to call `delete reply;`, rather than `free(reply);`, right?

You probably need `std::unique_ptr<xcb_intern_atom_reply_t, &free> const reply`?

Reply:
Something like that.

See https://github.com/MirServer/mir/pull/1725/commits/132134e77501ec894890393e8efe2096fd5ae5a0 for fir the solution I propose on another branch.